ts/Series/Contour/ContourSeries.ts

```diff
@@ -0,0 +1,915 @@
+/* *
+ *
+ *  (c) 2010-2025 Torstein Honsi
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ * */
+
+/// <reference types="@webgpu/types" />
+'use strict';
+
+
+/* *
+ *
+ *  Imports
+ *
+ * */
+
+import type Chart from '../../Core/Chart/Chart.js';
+import type ContourSeriesOptions from './ContourSeriesOptions';
+import type { DeepPartial } from '../../Shared/Types';
+import type SVGElement from '../../Core/Renderer/SVG/SVGElement.js';
+
+import Color from '../../Core/Color/Color.js';
+import ContourPoint from './ContourPoint.js';
+import contourShader from './contourShader.js';
+import ContourSeriesDefaults from './ContourSeriesDefaults.js';
+import Delaunay from '../../Shared/Delaunay.js';
+import SeriesRegistry from '../../Core/Series/SeriesRegistry.js';
+import U from '../../Core/Utilities.js';
+
+const {
+    seriesTypes: {
+        scatter: ScatterSeries
+    }
+} = SeriesRegistry;
+const {
+    diffObjects,
+    extend,
+    merge
+} = U;
+
+
+/* *
+ *
+ *  Class
+ *
+ * */
+
+export default class ContourSeries extends ScatterSeries {
+
+    /* *
+     *
+     * Static Properties
+     *
+     * */
+
+    public static defaultOptions = merge(
+        ScatterSeries.defaultOptions,
+        ContourSeriesDefaults
+    );
+
+
+    /* *
+     *
+     * Properties
+     *
+     * */
+
+    public data!: Array<ContourPoint>;
+
+    public points!: Array<ContourPoint>;
+
+    public options!: ContourSeriesOptions;
+
+    public context?: GPUCanvasContext | null;
+
+    public renderFrame?: () => void;
+
+    public renderPromise?: Promise<void>;
+
+    private foreignObject?: SVGForeignObjectElement;
+
+    private canvas?: HTMLCanvasElement;
+
+    private adapter?: GPUAdapter | null;
+
+    private device?: GPUDevice;
+
+    private extremesUniformBuffer?: GPUBuffer;
+
+    private valueExtremesUniformBuffer?: GPUBuffer;
+
+    private contourIntervalUniformBuffer?: GPUBuffer;
+
+    private contourOffsetUniformBuffer?: GPUBuffer;
+
+    private smoothColoringUniformBuffer?: GPUBuffer;
+
+    private showContourLinesUniformBuffer?: GPUBuffer;
+
+    private contourLineColorBuffer?: GPUBuffer;
+
+    private colorAxisStopsUniformBuffer?: GPUBuffer;
+
+    private colorAxisStopsCountUniformBuffer?: GPUBuffer;
+
+    private isInvertedUniformBuffer?: GPUBuffer;
+
+    /* Uniforms:
+     * - extremesUniform,
+     * - valueExtremesUniform,
+     * - contourInterval,
+     * - contourOffset,
+     * - smoothColoring,
+     * - showContourLines,
+     * - contourLineColor
+     * - colorAxisStops
+     * - colorAxisStopsCount
+     * - isInverted
+     */
+
+
+    /* *
+     *
+     * Methods
+     *
+     * */
+
+    public override init(chart: Chart, options: ContourSeriesOptions): void {
```

Comment:
I am in doubt if we should do this generic axis override. It works well when there is only a contour series in the chart, but results in some inconsistencies when multiple series are loaded. For example the axes updating when we add the contour dynamically, but doesn't revert after removing: https://jsfiddle.net/highcharts/o268xj0m/

It may be a little surprising to implementers that the series affect the axes this way. We also have an explanation problem when we change the defaults that are described in the axis docs.

Furthermore, we don't currently do this with any other series. For heatmaps for example, it would make equally sense as for contours. But for heatmaps we set `startOnTick`, `endOnTick` etc in the demos instead, where required.